dali/server/daserver.cpp

```diff
@@ -393,6 +393,53 @@ static StringBuffer &formatDaliRole(StringBuffer &out, unsigned __int64 role)
     return out.append(queryRoleName((DaliClientRole)role));
 }
 
+static IPropertyTree * getHelmLDAPConfiguration(const char * authMethod, IPropertyTree *appConfig)
+{
+    if (streq(authMethod, "none"))
+        return nullptr; //no security manager
+
+    const char *ldapAddress = appConfig->queryProp("@ldapAddress");
+    if (isEmptyString(ldapAddress))
+        throw MakeStringException(-1, "LDAP not configured (missing 'ldapAddress').  To run without security set auth=none");
+
+    //Get LDAP attributes from helm config
+    Owned<IPropertyTree> ldapConfig = appConfig->getPropTree("ldap");
+    ldapConfig->addProp("@ldapAddress", ldapAddress);
+
+    //Get default LDAP attributes from (azure_)ldap.yaml
+    StringBuffer ldapDefaultsFile(hpccBuildInfo.componentDir);
+    char sepchar = getPathSepChar(ldapDefaultsFile.str());
+    addPathSepChar(ldapDefaultsFile, sepchar).append("applications").append(sepchar).append("dali").append(sepchar).append("ldap").append(sepchar);
+    if (streq(authMethod, "ldap"))
+        ldapDefaultsFile.append("ldap.yaml");
+    else if (streq(authMethod, "azureldap"))
+        ldapDefaultsFile.append("azure_ldap.yaml");
+    else
+        throw MakeStringException(-1, "LDAP not configured (unrecognized 'auth=%s').  Set auth=none, ldap, or azureldap", authMethod);
+
+    //read properties from defaults file
+    Owned<IPropertyTree> ldapDefaults;
+    if (checkFileExists(ldapDefaultsFile))
+        ldapDefaults.setown(createPTreeFromYAMLFile(ldapDefaultsFile.str()));
+
+    //Append items from defaults file, but only if not already specified in app config
```

Comment:
I agree here also.  The final config should start with the default values and you can merge in the settings provided at the application level.  Instead of the other way around.  Generally there wont be many.  Those will then overwrite the defaults.  You'll never modify the appConfig/serverConfig.

The only thing I would point out is that what you are currently returning is the "ldap" section, so I think the code would have one more line:

```
Owned<ldapDefaults> = createPTreeFromYAMLFile(ldapDefaultsFile);
Owned<ldapConfig> = ldapDefaults->getPropTree("ldap");
mergePTree(ldapConfig, appConfig->queryPropTree("ldap");
```
You don't have to worry about skipping things that already exist or whatever because the appConfig side will always win.

Probably needs some pointer checking though.